nncf/torch/hardware/fused_patterns.py

```diff
@@ -13,49 +30,124 @@
+
+
+@PT_HW_FUSED_PATTERNS.register(PatternNames.LINEAR_ACTIVATIONS_BATCH_NORM_PERMUTATIONS)
+def create_linear_activation_batch_norm_permutation():
+    linear = linear_operations()
+    bn_activations_permutations = batch_norm_activations_permutations()
+    linear.join_patterns(bn_activations_permutations)
+    return linear
+
+
+@PT_HW_FUSED_PATTERNS.register(PatternNames.ARITHMETIC_BATCH_NORM_ACTIVATIONS_PERMUTATIONS)
+def create_arithmetic_batch_norm_activations_permutations():
+    arithmetic = arithmetic_operations()
+    bn_act_perm = batch_norm_activations_permutations()
+    arithmetic.join_patterns(bn_act_perm)
+    return arithmetic
+
+
+@PT_HW_FUSED_PATTERNS.register(PatternNames.GROUP_NORM_RELU)
+def create_group_norm_relu_operations():
+    group_norm = GraphPattern()
+    group_norm.add_node(**GROUP_NORMALIZATION_OPERATIONS)
+    relu = GraphPattern()
+    relu.add_node(**RELU_OPERATIONS)
+    group_norm.join_patterns(relu)
+    return group_norm
+
+
+@PT_HW_FUSED_PATTERNS.register(PatternNames.LINEAR_CONST_MULTIPLY)
+def create_linear_const_multiply():
+    return create_fc_conv_mul()
+
+
+def linear_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**LINEAR_OPERATIONS)
+    return pattern
+
+def arithmetic_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**ARITHMETIC_OPERATIONS)
+    return pattern
+
+
+def batch_norm_operations():
+    pattern = GraphPattern()
+    pattern.add_node(**BATCH_NORMALIZATION_OPERATIONS)
+    return pattern
+
+
+def activation_operations():
     atomic_activations = GraphPattern()
     atomic_activations.add_node(**ATOMIC_ACTIVATIONS_OPERATIONS)
     swish = create_swish_act()
     h_sigmoid = create_h_sigmoid_act()
     h_swish = create_h_swish_act()
-    activations = atomic_activations | swish | h_swish | h_sigmoid
-    retval.register(activations, 'ACTIVATIONS', match=False)
 
-    arithmetic_ops = GraphPattern()
-    arithmetic_ops.add_node(**ARITHMETIC_OPERATIONS)
-    retval.register(arithmetic_ops, ARITHMETIC_OPERATIONS['label'], match=False)
+    pattern = GraphPattern()
+    pattern.add_pattern_alternative(atomic_activations)
+    pattern.add_pattern_alternative(swish)
+    pattern.add_pattern_alternative(h_swish)
+    pattern.add_pattern_alternative(h_sigmoid)
+    return pattern
 
-    batch_norm_activations_permutation = batch_norm + activations | activations + batch_norm | batch_norm | activations
 
-    retval.register(linear_ops + arithmetic_ops, 'LINEAR + ARITHMETIC', match=True)
-    retval.register(linear_ops + batch_norm_activations_permutation, 'LINEAR + BN_ACT_PERM',
-                    match=True)
-    retval.register(batch_norm + activations, 'BN + ACTIVATIONS', match=True)
-    retval.register(activations + batch_norm, 'ACTIVATIONS + BN', match=True)
-    retval.register(arithmetic_ops + batch_norm_activations_permutation,
-                    'ARITHMETIC + BN_ACT_PERM', match=True)
+def batch_norm_activations_permutations():
```

Comment:
My suggestion is not to add so-called `permutations` of the patterns. This method contains the list of existing combinations such as `batch norm + activations`, `activations + batch norm` and etc., that should be described as BATCH_NORM_ACTIVATIONS and corresponding patterns.

Reply:
Done, please check